tools/clang/lib/SPIRV/SortDebugInfoVisitor.cpp

```diff
@@ -100,6 +100,12 @@ void SortDebugInfoVisitor::whileEachOperandOfDebugInstruction(
     if (!visitor(inst->getElementType()))
       break;
   } break;
+  case SpirvInstruction::IK_DebugTypeMatrix: {
+    SpirvDebugTypeMatrix *inst = dyn_cast<SpirvDebugTypeMatrix>(di);
+    assert(inst != nullptr);
+    if (!visitor(inst->getVectorType()))
+      break;
+  } break;
```

Comment:
I know you just repeated the pattern, but it would be better to:
- cast instead of dyn_cast (since you assert just after)
- don't check visitor() & break since you break just after anyway:

```
SpirvDebugTypeMatrix *inst = cast<SpirvDebugTypeMatrix>(di);
visitor(inst->getVectorType());
```